tests/test_modeling_common.py

```diff
@@ -700,6 +700,7 @@ def check_equal(loaded):
                 check_equal(load_state_dict(pt_checkpoint_path))
 
     def test_initialization(self):
+        torch.set_default_dtype(torch.float32)  # Other tests may change the default dtype, this test requires float32
```

Comment:
@ydshieh even after fixing `test_eager_matches_sdpa_generate`, this test was crashing. 

I think calling ` model_class.from_pretrained(..., torch_dtype=torch.float16)` changes the default dtype in torch, which was causing this test to fail (we were getting `inf` because numbers were out of range)

Reply:
Got it. This kind of effects of changing the states is quite bad.
Could you add `# TODO: ydshieh` here (or even opening a new issue) so I can come back to check?

Thanks!